### PR TITLE
Add package installer unit tests

### PR DESCRIPTION
**Pull request by aleksandr-dorofeev**
Please have a look at our [guidelines for contributions](https://github.com/microsoft/appcenter-sdk-android/blob/develop/CONTRIBUTING.md) and consider the following before you submit the PR:

* ~[ ] Has `CHANGELOG.md` been updated?~
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [x] Did you add unit tests?
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?
* [x] Did you check UI tests on the sample app? They are not executed on CI.

## Description
Added unit tests for new PackageInstaller implementation and fix old tests for old implementation.

## Related PRs or issues

[AB88757](https://msmobilecenter.visualstudio.com/Mobile-Center/_workitems/edit/88757)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,7 +10,7 @@
 ### App Center Distribute
 
 * **[Feature]** Remove the download manager task if the download doesn't start within 10 seconds.
-* **[Feature]** Replace install a new release from deprecated intent action [ACTION_INSTALL_PACKAGE](https://developer.android.com/reference/android/content/Intent#ACTION_INSTALL_PACKAGE) to `PackageInstaller` API.
+* **[Feature]** Replace installing a new release using the deprecated intent action [ACTION_INSTALL_PACKAGE](https://developer.android.com/reference/android/content/Intent#ACTION_INSTALL_PACKAGE) with the `PackageInstaller` API.
```
Comment (AnastasiaKubova) [inline CHANGELOG.md:13]: Why are these changes here? Please revert all changes from the main code and merge PR with the base branch.

File: sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/ReleaseInstallerListenerTest.java [hunk 2]
```diff
@@ -331,4 +331,22 @@ public void releaseInstallerProcessWhenWithContext() throws Exception {
         /* Verify that installer process was triggered in the Distribute again. */
         verify(mDistribute).notifyInstallProgress(eq(false));
     }
+
+    @Test
+    public void releaseInstallerHideDialogTwice() {
+        /* Start install process. */
```
Comment (AnastasiaKubova) [inline sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/ReleaseInstallerListenerTest.java:337]: empty line

File: sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/ReleaseInstallerListenerTest.java [hunk 3]
```diff
@@ -331,4 +331,22 @@ public void releaseInstallerProcessWhenWithContext() throws Exception {
...
+        mReleaseInstallerListener.hideInstallProgressDialog();
+
+        /* Try to hide dialog again */
+        mReleaseInstallerListener.hideInstallProgressDialog();
+
+        /* Verify that runnable was called once only. */
+        ArgumentCaptor<Runnable> runnable = ArgumentCaptor.forClass(Runnable.class);
+        verifyStatic(times(1));
+        HandlerUtils.runOnUiThread(runnable.capture());
+        runnable.getValue().run();
```
Comment (AnastasiaKubova) [inline sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/ReleaseInstallerListenerTest.java:350]: Do we really need these lines? What is it checking?

File: sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/DistributeWarnAlertSystemWindowsTest.java [hunk 4]
```diff
@@ -295,5 +300,70 @@ public void returningToDialogFromBackgroundWhenInstallerIsNull() {
...
+                .thenReturn(Mockito.mock(Notification.Builder.class, RETURNS_MOCKS));
+        Mockito.when(mContext.getSystemService(NOTIFICATION_SERVICE)).thenReturn(manager);
+
+        /* Notify about complete download. */
+        mReleaseDownloaderListener.onComplete(1L);
+
+        /* Verify that dialog was not shown. */
+        verify(mAlertWindowsDialog, never()).show();
+
+        /* Verify system alert window is not trying to display if activity is null */
```
Comment (AnastasiaKubova) [inline sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/DistributeWarnAlertSystemWindowsTest.java:327]: `Verify system alert window is not trying to display if activity is null` -> `Verify system alert window is not trying to display if activity is null.`

File: sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/DistributeWarnAlertSystemWindowsTest.java [hunk 5]
```diff
@@ -295,5 +300,70 @@ public void returningToDialogFromBackgroundWhenInstallerIsNull() {
...
 
+    @Test
+    public void showSettingsDialogWhenActivityIsNull() throws Exception {
+
+        /* Mock permission state for Android Q. */
+        when(InstallerUtils.isSystemAlertWindowsEnabled(any(Context.class))).thenReturn(false);
+        when(Settings.canDrawOverlays(any(Context.class))).thenReturn(false);
+
+        /* Set Activity NULL */
+        Distribute.getInstance().onActivityPaused(mActivity);
```
Comment (AnastasiaKubova) [inline sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/DistributeWarnAlertSystemWindowsTest.java:311]: Activity is not null here

File: sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/DistributeWarnAlertSystemWindowsTest.java [hunk 6]
```diff
@@ -295,5 +300,70 @@ public void returningToDialogFromBackgroundWhenInstallerIsNull() {
         AppCenterLog.debug(eq(LOG_TAG), eq("Installing couldn't start due to the release installer wasn't initialized."));
     }
 
+    @Test
+    public void showSettingsDialogWhenActivityIsNull() throws Exception {
+
+        /* Mock permission state for Android Q. */
```
Comment (AnastasiaKubova) [inline sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/DistributeWarnAlertSystemWindowsTest.java:306]: Let's also add info that you check it on Android Q to the test method.

File: sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/DistributeWarnAlertSystemWindowsTest.java [hunk 7]
```diff
@@ -295,5 +300,70 @@ public void returningToDialogFromBackgroundWhenInstallerIsNull() {
...
+
+        /* Verify system alert window is not trying to display if activity is null */
+        verifyStatic();
+        AppCenterLog.warn(eq(LOG_TAG), eq("The application is in background mode, the system alerts windows won't be displayed."));
+    }
+
+    @Test
+    public void needRefreshDialogWhenStartInstallation() throws Exception {
+
+        /* Mock permission state for Android Q. */
```
Comment (AnastasiaKubova) [inline sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/DistributeWarnAlertSystemWindowsTest.java:335]: Let's also add info that you check it on Android Q to the test method.

File: sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/DistributeWarnAlertSystemWindowsTest.java [hunk 8]
```diff
@@ -295,5 +300,70 @@ public void returningToDialogFromBackgroundWhenInstallerIsNull() {
...
+        when(InstallerUtils.isSystemAlertWindowsEnabled(any(Context.class))).thenReturn(false);
+        when(Settings.canDrawOverlays(any(Context.class))).thenReturn(false);
+
+        Mockito.when(DistributeUtils.getStoredDownloadState()).thenReturn(DOWNLOAD_STATE_NOTIFIED);
```
Comment (AnastasiaKubova) [inline sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/DistributeWarnAlertSystemWindowsTest.java:339]: empty line

File: sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/DistributeWarnAlertSystemWindowsTest.java [hunk 9]
```diff
@@ -295,5 +300,70 @@ public void returningToDialogFromBackgroundWhenInstallerIsNull() {
...
+    public void showSettingsDialogWhenActivityIsNull() throws Exception {
+
+        /* Mock permission state for Android Q. */
+        when(InstallerUtils.isSystemAlertWindowsEnabled(any(Context.class))).thenReturn(false);
+        when(Settings.canDrawOverlays(any(Context.class))).thenReturn(false);
+
+        /* Set Activity NULL */
+        Distribute.getInstance().onActivityPaused(mActivity);
+
+        Mockito.when(DistributeUtils.getStoredDownloadState()).thenReturn(DOWNLOAD_STATE_NOTIFIED);
```
Comment (AnastasiaKubova) [inline sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/DistributeWarnAlertSystemWindowsTest.java:313]: empty line

File: sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/DistributeWarnAlertSystemWindowsTest.java [hunk 10]
```diff
@@ -295,5 +300,70 @@ public void returningToDialogFromBackgroundWhenInstallerIsNull() {
...
+        whenNew(Notification.Builder.class).withAnyArguments()
+                .thenReturn(Mockito.mock(Notification.Builder.class, RETURNS_MOCKS));
+        Mockito.when(mContext.getSystemService(NOTIFICATION_SERVICE)).thenReturn(manager);
+
+        /* Notify about complete download. */
+        mReleaseDownloaderListener.onComplete(1L);
+
+        /* Start distribute with app secret NULL to make sure updateReleaseDetails is called on
+           startFromBackground.
+         */
```
Comment (AnastasiaKubova) [inline sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/DistributeWarnAlertSystemWindowsTest.java:352]: formatting

File: sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/DistributeWarnAlertSystemWindowsTest.java [hunk 11]
```diff
@@ -295,5 +300,70 @@ public void returningToDialogFromBackgroundWhenInstallerIsNull() {
...
+        Distribute.getInstance().onStarted(mContext, mChannel, null, null, true);
```
Comment (AnastasiaKubova) [inline sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/DistributeWarnAlertSystemWindowsTest.java:353]: You should call `onStarting(mAppCenterHandler)` before `onStarted`

File: sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/DistributeWarnAlertSystemWindowsTest.java [hunk 12]
```diff
@@ -295,5 +300,70 @@ public void returningToDialogFromBackgroundWhenInstallerIsNull() {
...
+        Distribute.getInstance().startFromBackground(mContext);
+
+        /* isMandatoryUpdate should be mocked to avoid completion installation workflow */
+        Mockito.when(mReleaseDetails.isMandatoryUpdate()).thenReturn(true);
+
+        mReleaseDownloaderListener.onComplete(1L);
```
Comment (AnastasiaKubova) [inline sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/DistributeWarnAlertSystemWindowsTest.java:359]: empty line

File: sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/DistributeWarnAlertSystemWindowsTest.java [hunk 13]
```diff
@@ -295,5 +300,70 @@ public void returningToDialogFromBackgroundWhenInstallerIsNull() {
...
+        /* Notify about complete download. */
+        mReleaseDownloaderListener.onComplete(1L);
+
+        /* Start distribute with app secret NULL to make sure updateReleaseDetails is called on
+           startFromBackground.
+         */
+        Distribute.getInstance().onStarted(mContext, mChannel, null, null, true);
+        Distribute.getInstance().startFromBackground(mContext);
+
+        /* isMandatoryUpdate should be mocked to avoid completion installation workflow */
```
Comment (AnastasiaKubova) [inline sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/DistributeWarnAlertSystemWindowsTest.java:356]: `isMandatoryUpdate should be mocked to avoid completion installation workflow` -> `Release details should be mandatory to avoid completion installation workflow.`

File: sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/InstallerUtilsTest.java [hunk 14]
```diff
@@ -34,8 +38,14 @@
 import java.io.IOException;
 import java.io.InputStream;
 import java.io.OutputStream;
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+
+@PrepareForTest({
+        PendingIntent.class,
+        Settings.class
+})
```
Comment (AnastasiaKubova) [inline sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/InstallerUtilsTest.java:48]: empty line?

File: sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/InstallerUtilsTest.java [hunk 15]
```diff
@@ -122,4 +132,61 @@ public void throwIOExceptionWhenTryToCreateSession() throws IOException {
         /* Verify that the session wasn't created. */
         verify(mMockPackageInstaller, never()).openSession(anyInt());
     }
+
+    @Test
+    public void isSystemAlertWindowsEnabledReturnsTrueIfBuildVersionLowerQ() throws Exception {
+        /* Mock SDK_INT to P */
+        setFinalStatic(Build.VERSION.class.getField("SDK_INT"), Build.VERSION_CODES.P);
```
Comment (AnastasiaKubova) [inline sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/InstallerUtilsTest.java:139]: please add an empty line

File: sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/InstallerUtilsTest.java [hunk 16]
```diff
@@ -122,4 +132,61 @@ public void throwIOExceptionWhenTryToCreateSession() throws IOException {
...
+        when(Settings.canDrawOverlays(any(Context.class))).thenReturn(true);
+
+        /* Check if system alert windows is enabled */
+        assertTrue(InstallerUtils.isSystemAlertWindowsEnabled(mContext));
+    }
+
+    @Test
+    public void isSystemAlertWindowsEnabledReturnsFalseIfBuildVersionHigherQAndCanDrawOverlay() throws Exception {
+        /* Mock SDK_INT to P */
+        setFinalStatic(Build.VERSION.class.getField("SDK_INT"), Build.VERSION_CODES.Q);
```
Comment (AnastasiaKubova) [inline sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/InstallerUtilsTest.java:160]: please add an empty line

File: sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/InstallerUtilsTest.java [hunk 17]
```diff
@@ -122,4 +132,61 @@ public void throwIOExceptionWhenTryToCreateSession() throws IOException {
...
+        mockStatic(Settings.class);
+        when(Settings.canDrawOverlays(any(Context.class))).thenReturn(true);
+
+        /* Check if system alert windows is enabled */
+        assertTrue(InstallerUtils.isSystemAlertWindowsEnabled(mContext));
+    }
+
+    @Test
+    public void isSystemAlertWindowsEnabledReturnsFalseIfBuildVersionHigherQAndCannotDrawOverlay() throws Exception {
+        /* Mock SDK_INT to P */
```
Comment (AnastasiaKubova) [inline sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/InstallerUtilsTest.java:172]: please add an empty line

File: sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/InstallerUtilsTest.java [hunk 18]
```diff
@@ -122,4 +132,61 @@ public void throwIOExceptionWhenTryToCreateSession() throws IOException {
...
+        /* Set canDrawOverlays to false */
+        mockStatic(Settings.class);
+        when(Settings.canDrawOverlays(any(Context.class))).thenReturn(false);
+
+        /* Check if system alert windows is enabled */
+        assertFalse(InstallerUtils.isSystemAlertWindowsEnabled(mContext));
+    }
+
+    static void setFinalStatic(Field field, Object newValue) throws Exception {
+        field.setAccessible(true);
```
Comment (AnastasiaKubova) [inline sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/InstallerUtilsTest.java:184]: remove empty lines or add comments

File: sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/InstallerUtilsTest.java (hunk 17 above)
Comment (AnastasiaKubova) [inline sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/InstallerUtilsTest.java:172]: Why P if you mock `Build.VERSION_CODES.Q`?

File: sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/InstallerUtilsTest.java [hunk 19]
```diff
@@ -122,4 +132,61 @@ public void throwIOExceptionWhenTryToCreateSession() throws IOException {
...
+        assertTrue(InstallerUtils.isSystemAlertWindowsEnabled(mContext));
+    }
+
+    @Test
+    public void isSystemAlertWindowsEnabledReturnsFalseIfBuildVersionHigherQAndCannotDrawOverlay() throws Exception {
+        /* Mock SDK_INT to P */
+        setFinalStatic(Build.VERSION.class.getField("SDK_INT"), Build.VERSION_CODES.Q);
+
+        /* Set canDrawOverlays to false */
+        mockStatic(Settings.class);
```
Comment (AnastasiaKubova) [inline sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/InstallerUtilsTest.java:176]: `Set canDrawOverlays to false` -> `Set canDrawOverlays to false.`

File: sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/InstallerUtilsTest.java [hunk 20]
```diff
@@ -122,4 +132,61 @@ public void throwIOExceptionWhenTryToCreateSession() throws IOException {
...
+        when(Settings.canDrawOverlays(any(Context.class))).thenReturn(false);
+
+        /* Check if system alert windows is enabled */
+        assertFalse(InstallerUtils.isSystemAlertWindowsEnabled(mContext));
```
Comment (AnastasiaKubova) [inline sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/InstallerUtilsTest.java:180]: `Check if system alert windows is enabled` -> `Check if system alert windows is enabled`

File: sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/InstallerUtilsTest.java [hunk 21]
```diff
@@ -122,4 +132,61 @@ public void throwIOExceptionWhenTryToCreateSession() throws IOException {
...
+        * */
+        when(Settings.canDrawOverlays(any(Context.class))).thenReturn(true);
+
+        /* Check if system alert windows is enabled */
+        assertTrue(InstallerUtils.isSystemAlertWindowsEnabled(mContext));
+    }
+
+    @Test
+    public void isSystemAlertWindowsEnabledReturnsFalseIfBuildVersionHigherQAndCanDrawOverlay() throws Exception {
+        /* Mock SDK_INT to P */
```
Comment (AnastasiaKubova) [inline sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/InstallerUtilsTest.java:159]: Why `P` if you mock `Build.VERSION_CODES.Q`?

File: sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/InstallerUtilsTest.java [hunk 22]
```diff
@@ -122,4 +132,61 @@ public void throwIOExceptionWhenTryToCreateSession() throws IOException {
...
+        setFinalStatic(Build.VERSION.class.getField("SDK_INT"), Build.VERSION_CODES.Q);
+
+        /* Set canDrawOverlays to false */
```
Comment (AnastasiaKubova) [inline sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/InstallerUtilsTest.java:162]: `Set canDrawOverlays to false` -> `Set canDrawOverlays to false.`

File: sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/InstallerUtilsTest.java [hunk 23]
```diff
@@ -122,4 +132,61 @@ public void throwIOExceptionWhenTryToCreateSession() throws IOException {
...
+        mockStatic(Settings.class);
+        when(Settings.canDrawOverlays(any(Context.class))).thenReturn(true);
+
+        /* Check if system alert windows is enabled */
```
Comment (AnastasiaKubova) [inline sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/InstallerUtilsTest.java:166]: `Check if system alert windows is enabled` -> `Check if system alert windows is enabled.`

File: sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/InstallerUtilsTest.java [hunk 24]
```diff
@@ -122,4 +132,61 @@ public void throwIOExceptionWhenTryToCreateSession() throws IOException {
...
+        assertTrue(InstallerUtils.isSystemAlertWindowsEnabled(mContext));
+
+        /* Set canDrawOverlays to true and make sure it doesn't affect
+        * the return value of isSystemAlertWindowsEnabled.
+        * */
+        when(Settings.canDrawOverlays(any(Context.class))).thenReturn(true);
+
+        /* Check if system alert windows is enabled */
```
Comment (AnastasiaKubova) [inline sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/InstallerUtilsTest.java:153]: `Check if system alert windows is enabled` -> `Check if system alert windows is enabled.`

File: sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/InstallerUtilsTest.java [hunk 25]
```diff
@@ -122,4 +132,61 @@ public void throwIOExceptionWhenTryToCreateSession() throws IOException {
...
+    @Test
+    public void isSystemAlertWindowsEnabledReturnsTrueIfBuildVersionLowerQ() throws Exception {
+        /* Mock SDK_INT to P */
+        setFinalStatic(Build.VERSION.class.getField("SDK_INT"), Build.VERSION_CODES.P);
+
+        /* Set canDrawOverlays to false */
+        mockStatic(Settings.class);
+        when(Settings.canDrawOverlays(any(Context.class))).thenReturn(false);
+
+        /* Check if system alert windows is enabled */
```
Comment (AnastasiaKubova) [inline sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/InstallerUtilsTest.java:145]: `Check if system alert windows is enabled` -> `Check if system alert windows is enabled.`

File: sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/InstallerUtilsTest.java [hunk 26]
```diff
@@ -122,4 +132,61 @@ public void throwIOExceptionWhenTryToCreateSession() throws IOException {
         /* Verify that the session wasn't created. */
         verify(mMockPackageInstaller, never()).openSession(anyInt());
     }
+
+    @Test
+    public void isSystemAlertWindowsEnabledReturnsTrueIfBuildVersionLowerQ() throws Exception {
+        /* Mock SDK_INT to P */
+        setFinalStatic(Build.VERSION.class.getField("SDK_INT"), Build.VERSION_CODES.P);
+
+        /* Set canDrawOverlays to false */
```
Comment (AnastasiaKubova) [inline sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/InstallerUtilsTest.java:141]: `Set canDrawOverlays to false` -> `Set canDrawOverlays to false.`

File: sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/InstallerUtilsTest.java [hunk 27]
```diff
@@ -122,4 +132,61 @@ public void throwIOExceptionWhenTryToCreateSession() throws IOException {
...
+        mockStatic(Settings.class);
+        when(Settings.canDrawOverlays(any(Context.class))).thenReturn(false);
+
+        /* Check if system alert windows is enabled */
+        assertTrue(InstallerUtils.isSystemAlertWindowsEnabled(mContext));
+
+        /* Set canDrawOverlays to true and make sure it doesn't affect
```
Comment (AnastasiaKubova) [inline sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/InstallerUtilsTest.java:148]: let's split this comment:
`Set canDrawOverlays to true.`

and before checking
`Make sure that it does not affect the result of the permission check.

File: sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/DistributeWarnAlertSystemWindowsTest.java [hunk 28]
```diff
@@ -295,5 +300,70 @@ public void returningToDialogFromBackgroundWhenInstallerIsNull() {
...
+        Mockito.when(mReleaseDetails.isMandatoryUpdate()).thenReturn(true);
+
+        mReleaseDownloaderListener.onComplete(1L);
+
+        /* Verify that dialog was shown once only. */
+        verify(mAlertWindowsDialog).show();
+
+        /* Verify system alert window is not trying to display if activity is null */
```
Comment (AnastasiaKubova) [inline sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/DistributeWarnAlertSystemWindowsTest.java:364]: `Verify system alert window is not trying to display if activity is null` -> `Verify system alert window is not trying to display if activity is null.`

File: sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/InstallerUtilsTest.java (hunk 15 above)
Comment (AnastasiaKubova) [reply sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/InstallerUtilsTest.java:139]: Let's use the minimum supported platform `Lollipop` instead of `P`.

File: sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/DistributeTest.java [hunk 29]
```diff
@@ -845,6 +855,141 @@ public void checkUpdateReleaseAfterInterruptDownloading() {
...
+        /* Disable Distribute. */
+        Distribute.setEnabled(false);
+
+        /* Check that receiver was unregistered. */
+        verify(mActivity).unregisterReceiver(Matchers.<BroadcastReceiver>any());
+    }
+
+    @Test
+    public void checkRegisterReceiverWhenActivityRegisterReceiverThrowsException() {
+        /* Mock throwing exception. */
```
Comment (AnastasiaKubova) [inline sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/DistributeTest.java:895]: please add empty line

File: sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/ReleaseInstallerListenerTest.java (hunk 3 above)
Comment (DmitriyKirakosyan) [reply sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/ReleaseInstallerListenerTest.java:350]: This checks that the code of hiding dialog is invoking only once. `HandlerUtils.runOnUiThread(new Runnable() {` in `hideInstallProgressDialog`.

File: sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/InstallerUtilsTest.java [hunk 30]
```diff
@@ -0,0 +1,204 @@
...
+        when(Settings.canDrawOverlays(any(Context.class))).thenReturn(false);
+
+        /* Verify that system alert windows is not enabled */
+        assertFalse(InstallerUtils.isSystemAlertWindowsEnabled(mContext));
+    }
+
+    /**
+     * This is used ot set a specific Build.VERSION.SDK_INT for tests.
+     * @param field
+     * @param newValue
```
Comment (AnastasiaKubova) [inline sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/InstallerUtilsTest.java:194]: please add description
